src/harness/reference_models/fss_purge/fss_purge.py

```diff
@@ -0,0 +1,332 @@
+import numpy as np
+
+from collections import namedtuple
+from reference_models.antenna import antenna
+from reference_models.geo import terrain, vincenty
+from reference_models.propagation import wf_itm
+
+# Definition of constants and data types for FSS purge model.
+# Frequency used in propagation model (in MHz) [R2-SGN-04]
+FREQ_PROP_MODEL = 3625.0
+
+# Reference Channel bandwidth for FSS OOBE calculation
+REF_BW = 5.e6
+
+# One Mega Hertz
+ONE_MHZ = 1.e6
+
+# Define FSS information tuple containing the parameters necessary for antenna
+# and propagation calculation as listed in the test spec
+FssProtectionPoint = namedtuple('FssProtectionPoint',
+                           ['latitude',
+                            'longitude',
+                            'height_agl',
+                            'max_gain_dbi',
+                            'pointing_azimuth',
+                            'pointing_elevation',
+                            'weight_1',
+                            'weight_2'])
+
+def getFssInfo(fss_record):
+  """ Create a FssProtectionPoint tuple from the fss_record.
+
+  Args:
+    fss_record: A single FSS record dictionary.
+  Returns:
+    FssProtectionPoint: Tuple holding details about FSS needed for interference calculation.
+  """
+
+  # Extract fields from FSS record. 
+  fss_latitude = fss_record['deploymentParam'][0]['installationParam']['latitude']
+  fss_longitude = fss_record['deploymentParam'][0]['installationParam']['longitude']
+  fss_height = fss_record['deploymentParam'][0]['installationParam']['height']
+  fss_height_type = fss_record['deploymentParam'][0]['installationParam']['heightType']
+  fss_azimuth = fss_record['deploymentParam'][0]['installationParam']['antennaAzimuth']
+  fss_pointing_elevation = fss_record['deploymentParam'][0]['installationParam']['elevation']
+  fss_max_gain_dbi = fss_record['deploymentParam'][0]['installationParam']['antennaGain']
+  fss_weight_1 = fss_record['deploymentParam'][0]['weight1']
+  fss_weight_2 = fss_record['deploymentParam'][0]['weight2']
+  # Convert the height to AGL if it was AMSL type.
+  if fss_height_type == 'AMSL':
+     fss_height_agl = convertAmslToAgl(fss_latitude, fss_longitude, fss_height)
+  else:
+     fss_height_agl = fss_height
+
+  # Populate FssProtectionPoint tuple.
+  fss_entity = FssProtectionPoint(latitude=fss_latitude,
+                                  longitude=fss_longitude,
+                                  height_agl=fss_height_agl,
+                                  max_gain_dbi=fss_max_gain_dbi,
+                                  pointing_azimuth=fss_azimuth,
+                                  pointing_elevation=fss_pointing_elevation,
+                                  weight_1=fss_weight_1,
+                                  weight_2=fss_weight_2)
+  return fss_entity
+
+def getNeighbouringCbsdsToFss(cbsd_list, fss_record):
+  """ Get the list of all CBSDs that lie within 40 KMs of FSS.
+
+  Args:
+    cbsd_list: List of CBSDs.

```

Comment:
Registration requests? CbsdData dictionaries? CBSD objects? Something else?

Reply:
It is   CbsdData dictionary. The text has been updated.